src/GuzzleRequestAdapter.php

```diff
@@ -497,7 +497,7 @@ private function getRootParseNode(ResponseInterface $response, SpanInterface $sp
         $scope = $rootParseNodeSpan->activate();
         try {
             if (!$response->hasHeader(RequestInformation::$contentTypeHeader)) {
-                return null;
+                throw new NoContentTypeHeaderException("No response content type header for deserialization", $response);
```

Comment:
This would be diverging from the other languages implementations.
returning a null value instead of a parse node is perfectly acceptable in that scenario:
1. the API described it **could** return something we can parse
2. at runtime the API ended up returning nothing (201/204/etc... scenarios) or a content type that's not parsable (octet stream, etc...)
3. the client returns null to the application instead of failing.

Starting to throw exceptions in that scenario would break a whole lot of applications.

Reply:
seems my PR was based on outdated code then. I'll be researching some more about why this is an issue using the Microsoft graph API library as this lib uses this as a sub-lib. I see the current release features the return null line, yet in the version used by the graph API library, this line is a RuntimeException. I'll be bringing my shenanigans over there. feel free do decline PR